optuna/integration/allennlp.py

```diff
@@ -9,17 +10,156 @@
     import _jsonnet
+    from allennlp.training import EpochCallback
 else:
-    EpochCallback = object  # NOQA
+    # `allennlp.training.EpochCallback` is a subclass of `Registerable`
+    # https://docs.allennlp.org/master/api/training/trainer/#epochcallback
+    # On the other hand, `EpochCallback` defined here is not `Registerable`,
+    # which causes a mypy checking feailure.
+    class EpochCallback:  # type: ignore
+        """Stub for EpochCallback.
+
+        This class is introduced for documentation CI.
+
+        """
+
+        @classmethod
+        def register(cls: Any, *args: Any, **kwargs: Any) -> Callable:
+            def wrapper(subclass: Any, *args: Any, **kwargs: Any) -> None:
+                return subclass
+
+            return wrapper
+
+
+def _add_ppid(key: str) -> str:
+    """Add the ID of parent process to given string.
+
+    User might want to launch multiple studies that uses `AllenNLPExecutor`.
+    Because `AllenNLPExecutor` uses environment variables for communicating
+    between a parent process and a child process. A parent process creates a study,
+    defines a search space, and a child process trains a AllenNLP model by
+    `allennlp.commands.train.train_model`. If multiple processes use `AllenNLPExecutor`,
+    the one's configuration could be loaded in the another's configuration.
+    To avoid this hazard, we add ID of a parent process to each key of
+    environment variables.
+
+    """
+    ppid = os.getppid()
+    return "{}_{}".format(ppid, key)
+
+
+def _create_pruner() -> Optional[optuna.pruners.BasePruner]:
+    """Restore a pruner which is defined in `create_study`.
+
+    `AllenNLPPruningCallback` is launched as a sub-process of
+    a main script that defines search spaces.
+    An instance cannot be passed directly from the parent process
+    to its sub-process. For this reason, we set information about
+    pruner as environment variables and load them and
+    re-create the same pruner in `AllenNLPPruningCallback`.
+
+    """
+    pruner_class = os.getenv(_add_ppid("OPTUNA_ALLENNLP_PRUNER_CLASS"))
+    if pruner_class is None:
+        return None
+
+    pruner_params = _get_environment_variables_for_pruner()
+    pruner = getattr(optuna.pruners, pruner_class, None)
+
+    if pruner is None:
+        return None
+
+    return pruner(**pruner_params)
+
+
+def _infer_and_cast(value: Optional[str]) -> Optional[Union[str, int, float, bool]]:
+    """Infer and cast a string to desired types.
+
+    We are only able to set strings as environment variables.
+    However, parameters of a pruner could be integer, float,
+    boolean, or else. We infer and cast environment variables
+    to desired types.
+
+    """
+    if value is None:
+        return None
+
+    try:
+        return int(value)
+    except ValueError:
+        try:
+            return float(value)
+        except ValueError:
+            if value == "True":
+                return True
+            elif value == "False":
```

Comment:
A nit. Since we're returning if the preceding `if`.
```suggestion
            if value == "False":
```